src/main/java/org/terasology/multiBlock2/recipe/LayeredMultiBlockRecipe.java

```diff
@@ -65,23 +68,23 @@ public T detectFormingMultiBlock(Vector3i location) {
         return null;
     }
 
-    private T processDetectionForLayer(int layerIndex, Vector3i basePosition) {
+    private T processDetectionForLayer(int layerIndex, Vector3ic basePosition) {
         BlockEntityRegistry blockEntityRegistry = CoreRegistry.get(BlockEntityRegistry.class);
         LayerDefinition layerDefinition = layerDefinitions.get(layerIndex);
         Predicate<EntityRef> entityFilter = layerDefinition.entityFilter;
-        int minX = getLastMatchingInDirection(blockEntityRegistry, entityFilter, basePosition, Vector3i.east()).x;
-        int maxX = getLastMatchingInDirection(blockEntityRegistry, entityFilter, basePosition, Vector3i.west()).x;
-        int minZ = getLastMatchingInDirection(blockEntityRegistry, entityFilter, basePosition, Vector3i.south()).z;
-        int maxZ = getLastMatchingInDirection(blockEntityRegistry, entityFilter, basePosition, Vector3i.north()).z;
+        int minX = getLastMatchingInDirection(entityFilter, basePosition, Direction.RIGHT.asVector3i()).x;
+        int maxX = getLastMatchingInDirection(entityFilter, basePosition, Direction.LEFT.asVector3i()).x;
+        int minZ = getLastMatchingInDirection(entityFilter, basePosition, Direction.DOWN.asVector3i()).z;
+        int maxZ = getLastMatchingInDirection(entityFilter, basePosition, Direction.UP.asVector3i()).z;
```

Comment:
`north` and `south` are not `UP`/`DOWN`.
```suggestion
        int minZ = getLastMatchingInDirection(entityFilter, basePosition, Direction.BACKWARD.asVector3i()).z;
        int maxZ = getLastMatchingInDirection(entityFilter, basePosition, Direction.FORWARD.asVector3i()).z;
```